Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -796,18 +796,23 @@ std::pair<AppendPipeline::CreateTrackResult, AppendPipeline::Track*> AppendPipel
     }
 
     size_t newTrackIndex = m_tracks.size();
+    size_t webkitTrackIndex = std::count_if(m_tracks.begin(), m_tracks.end(), [streamType](std::unique_ptr<Track>& other) -> bool {
+        return other->streamType == streamType;
+    });
+
     TrackID preferredTrackId = getStreamIdFromPad(demuxerSrcPad).value_or((static_cast<TrackID>(newTrackIndex)));
     auto assignedTrackId = m_sourceBufferPrivate.tryRegisterTrackId(preferredTrackId);
     TrackID trackId = assignedTrackId.value_or(preferredTrackId);
 
     if (!assignedTrackId)
         GST_WARNING_OBJECT(pipeline(), "Failed to register track ID %" PRIu64 ", this could cause ID collisions", preferredTrackId);
 
-    GST_DEBUG_OBJECT(pipeline(), "Creating new AppendPipeline::Track with id '%" PRIu64 "'", trackId);
+    GST_DEBUG_OBJECT(pipeline(), "Creating new AppendPipeline::Track with type %s, index %" PRIu64" and id '%" PRIu64 "'", streamTypeToString(streamType), webkitTrackIndex, trackId);
     m_tracks.append(makeUnique<Track>(trackId, streamType, parsedCaps, presentationSize));
     Track& track = *m_tracks.at(newTrackIndex);
+
     track.initializeElements(this, GST_BIN(m_pipeline.get()));
-    track.webKitTrack = makeWebKitTrack(newTrackIndex, trackId);
+    track.webKitTrack = makeWebKitTrack(webkitTrackIndex, trackId);
```

Comment:
This will fail inside `makeWebKitTrack()`, as the first line of that function is:

```
Track& appendPipelineTrack = *m_tracks.at(trackIndex);
```

I'm also not sure it makes sense to use the indices of the `AppendPipeline::Track` in `m_tracks` for anything outside of `AppendPipeline`, but I can also confirm by looking at 569fdc337c908665da86de50acf2adcf89d7624e that it is my fault, as I was already doing that in the original multi-track support patch.

Reply:
Wouldn't the most straightforward solution here be to just pass the `Track&` as another argument to `makeWebKitTrack()`?

I'm also not sure what you mean by using the indices outside `AppendPipeline`?

Reply:
Assuming nothing else is broken, yes.

> I'm also not sure what you mean by using the indices outside AppendPipeline?

I mean how the `trackIndex` argument of `makeWebKitTrack()` is passed to the constructor of VideoTrackPrivateGStreamer etc.

Reply:
FWIW I was able to confirm the indices for TextTrackPrivateGStreamer and TextTrackPrivateBase are both the same, and per-type.